app.rb

```diff
@@ -2,29 +2,46 @@
 require 'bundler/setup'
 require_relative 'publisher.rb'
 require_relative 'subscriber.rb'
+require_relative 'q_it_errors'
 
-def run_service(options)
+SERVICES = ['publish', 'subscribe']
+
+def run_service(service, options)
+  if valid_options?(options)
```

Comment:
Not a good idea to have this as a separate method. And within that method `valid_options?` we have a `switch` statement similar to what is in this method so it's kind of redundant. Have the validation method be called in each individual switch. You anyway do have a begin rescue (L45) - so even if an exception is raised due to validations you can catch it and raise accordingly. L45 could be your single exit, and you can control handling exceptions gracefully if the need arises.

Reply:
was thinking it is getting redundant.. but not sure if putting it in there would be good or something else.. so left it at that for the meanwhile.. will change it..